### DISCUSSION TITLE
Add notes to marked lines

### DISCUSSION THREAD

**Discussion by jmichaelsturm** [variar/klogg#535] [Ideas]
Let me start by saying how much I love using klogg, and I think this is a great tool and project. I want to start contributing to the project as my time allows.

I am interested in creating a new feature (described below) if there would be interest and if this fits into the vision for how others use the app. I would want it to fit within the existing interface seamlessly and be "completely optional" if the user wishes to use it.

When I use klogg, I am always trying to figure out what went wrong in a particular log file. This is a painstaking process of discovery. Marking lines of interest is crucial to "telling the story" of what went wrong, but I would love to be able to add "notes" which would be visible in the main window and also the filtered window. These notes will allow me to document my findings about the log as I study it.

My vision for this would be something like an "enhanced mark" or a "note mark". When the user selects "Add note" from a right-click or from a shortcut key, the line will be marked, and a new dialog box will be shown for the user to type in a short one-line text note about the marked line. This is similar to the discussion on issue #274.

- If a marked line also includes a note, it will be displayed in both the main and filtered windows.
- If a marked line that includes a note is selected, the line text and the note can be copied to the scratchpad or the clipboard.
- Notes associated with a marked line can be deleted through a right-click action or a key shortcut.
- Notes for open log files would be preserved between sessions, but would not be written to the original log file.

Please let me know if this is something that is feasible to implement and would be a welcome addition to the tool.

Thanks,
Mike

**Comment by variar** [comment]
Hi, I'm really sorry for not picking up this discussion earlier. Switchig jobs and moving to another country don't leave much time for klogg. I'd love to have this feature. The scratchpad window is there partially for this exact issue of adding notes to interesting lines. The part of adding note should be rather straightforward to implement. Displaying notes in main/filtered window might require more work as klogg uses custom rendering for these widgets
